fvm.go

```diff
@@ -0,0 +1,206 @@
+//go:build cgo && (amd64 || arm64 || riscv64)
+// +build cgo
+// +build amd64 arm64 riscv64
+
+package ffi
+
+// #cgo linux LDFLAGS: ${SRCDIR}/libfilcrypto.a -Wl,-unresolved-symbols=ignore-all
+// #cgo darwin LDFLAGS: ${SRCDIR}/libfilcrypto.a -Wl,-undefined,dynamic_lookup
+// #cgo pkg-config: ${SRCDIR}/filcrypto.pc
+// #include "./filcrypto.h"
+import "C"
+import (
+	"context"
+	gobig "math/big"
+	"runtime"
+	"unsafe"
+
+	"github.com/filecoin-project/filecoin-ffi/cgo"
+	"github.com/filecoin-project/filecoin-ffi/generated"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+	"github.com/filecoin-project/go-state-types/network"
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+)
+
+type FVM struct {
+	executor unsafe.Pointer
+}
+
+const (
+	applyExplicit = iota
+	applyImplicit
+)
+
+type FVMOpts struct {
+	FVMVersion uint64
+	Externs    cgo.Externs
+
+	Epoch          abi.ChainEpoch
+	BaseFee        abi.TokenAmount
+	BaseCircSupply abi.TokenAmount
+	NetworkVersion network.Version
+	StateBase      cid.Cid
+}
+
+// CreateFVM creates a new FVM instance.
+func CreateFVM(opts *FVMOpts) (*FVM, error) {
+	baseFeeHi, baseFeeLo, err := splitBigInt(opts.BaseFee)
+	if err != nil {
+		return nil, xerrors.Errorf("invalid basefee: %w", err)
+	}
+	baseCircSupplyHi, baseCircSupplyLo, err := splitBigInt(opts.BaseCircSupply)
+	if err != nil {
+		return nil, xerrors.Errorf("invalid circ supply: %w", err)
+	}
+
+	exHandle := cgo.Register(context.TODO(), opts.Externs)
```

Comment:
pass the context to `CreateFVM`?

Reply:
Let's reconsider this later. `NewVM` in lotus _does_ take a context, but it's a context for the construction of the VM.